creator-node/src/premiumContent/helpers.ts

```diff
@@ -1,5 +1,62 @@
 import { signatureHasExpired } from '../apiSigning'
 import { PremiumContentSignatureData, PremiumContentType } from './types'
+import { getRegisteredDiscoveryNodes } from '../utils/getRegisteredDiscoveryNodes'
+import { Redis } from 'ioredis'
+import type Logger from 'bunyan'
+
+const models = require('../models')
+
+const PREMIUM_CONTENT_SIGNATURE_MAX_TTL_MS = 6 * 60 * 60 * 1000 // 6 hours
+
+export async function isCIDForPremiumTrack(cid: string): Promise<
+  | {
+      trackId: null
+      isPremium: false
+    }
+  | {
+      trackId: number
+      isPremium: boolean
+    }
+> {
+  const track = await models.sequelize.query(
+    `select t.* from "Tracks" t, "Files" f
+      where f."multihash" = :cid
+      and f."type" in ('track', 'copy320')
+      and f."trackBlockchainId" = t."blockchainId"`,
```

Comment:
1. any benchmark on avg query duration on prod?
2. plus1 on indices
3. this basically is a no-op of type is not track/copy320 - can we document at top of function or in the caller?

there's lots of nested logic and ideally we could document the various conditions in one place

Reply:
I've seen the query duration be anywhere from 1-6ms to 30ms. Assumption is that it's in the single digit ms if the query is cached, otherwise in the 2 digits.

Reply:
no way 30ms on prod

Reply:
@SidSethi psqling into the DB and running a query like `select t.* from "Tracks" t inner join "Files" f on f."trackBlockchainId"=t."blockchainId" where f."multihash"='QmZa6xTbrM9QbEa9HrkbLq2J3Ewu2WWQdoahqW27skkU2u' and f."type" in ('track', 'copy320');` took 30ms for me

Reply:
wow amazing